network/src/protocols/discovery/addr.rs

```diff
@@ -52,50 +46,36 @@ pub trait AddressManager {
 
 // bitcoin: bloom.h, bloom.cpp => CRollingBloomFilter
 pub struct AddrKnown {
-    max_known: usize,
-    addrs: HashSet<RawAddr>,
-    addr_times: HashMap<RawAddr, Instant>,
-    time_addrs: BTreeMap<Instant, RawAddr>,
+    filters: StableBloomFilter<DefaultBuildHashKernels<RandomState>>,
 }
 
 impl AddrKnown {
     pub(crate) fn new(max_known: usize) -> AddrKnown {
         AddrKnown {
-            max_known,
-            addrs: HashSet::default(),
-            addr_times: HashMap::default(),
-            time_addrs: BTreeMap::default(),
+            filters: StableBloomFilter::new(
+                max_known,
```

Comment:
The name `max_known` no longer makes sense, it is the number of buckets in the filter.